.github/CONTRIBUTING.md

```diff
@@ -1,16 +1,110 @@
+  * [Fetching the Sources From GitHub](#fetching-the-sources-from-github)
+  * [Linting and Testing OpenCollar with Your Changes](#Linting-and-Testing-OpenCollar-with-Your-Changes)
+
+## Asking Support Questions
+
+We have active [discussion channels](https://discord.gg/WUmBMemYqf) where users and developers can ask questions.
+Please don't use the GitHub issue tracker to ask general questions.
+
+## Reporting Issues
+
+If you believe you have found a defect in OpenCollar or its documentation, use
+the GitHub issue tracker to report
+the problem to the maintainers. Please do a quick search of existing issues before making a new one. When reporting the issue, please provide the version of OpenCollar in use (Help > About).
+
+- [OpenCollar Issues · OpenCollarTeam/OpenCollar](https://github.com/OpenCollarTeam/OpenCollar/issues)
+- [OpenCollar Documentation · OpenCollarTeam/opencollarteam.github.io](https://github.com/OpenCollarTeam/opencollarteam.github.io)
+
+## Code Contribution
+
+As OpenCollar's user base continues to grow, any new functionality must:
+
+* be useful to many.
+* fit naturally into _what OpenCollar does best._
+* strive not to break existing functionality
+* close or update an open [OpenCollar Issue](https://github.com/OpenCollarTeam/OpenCollar/issues)
+
+If the feature is of some complexity, the contributor is expected to maintain and support the new feature in the future (answer questions in-world or in discord, fix any bugs etc.).
+
+It is recommended to open up a discussion on the [Issues Tab](https://github.com/OpenCollarTeam/OpenCollar/issues) to get feedback on your idea before you begin. 
+
+Any non-trivial code change needs to update an open [issue](https://github.com/OpenCollarTeam/OpenCollar/issues). A non-trivial code change without an issue reference with one of the labels `bug` or `enhancement` will not be merged.
+
+**Bug fixes are, of course, always welcome.**
+
+## Submitting Patches
+
+The [OpenCollar](https://github.com/OpenCollarTeam/OpenCollar) Team welcomes all contributors and contributions regardless of skill or experience level. If you are interested in helping with the project, we will help you with your contribution.
+
+### Code Contribution Guidelines
+
+Because we want to create the best possible product for our users and the best contribution experience for our developers, we have a set of guidelines which ensure that all contributions are acceptable. The guidelines are not intended as a filter or barrier to participation. If you are unfamiliar with the contribution process, the [OpenCollar](https://github.com/OpenCollarTeam/OpenCollar) team will help you and teach you how to bring your contribution in accordance with the guidelines.
+
+To make the contribution process as seamless as possible, we ask for the following:
+
+* Go ahead and fork the project and make your changes.  We encourage pull requests to allow for review and discussion of code changes.
```

Comment:
This does not differentiate between staff and external contributors.  Staff being defined as anyone invited to the github organization with commit rights, and a member of the Staff SL group.  External contributor being defined as someone not with those commit rights.  The guidelines discussed in discord make this a very rigid process that discourages contribution from staff members, due to the requirement of a review for staff contributions.  I agree that if there is more than 1 active staff developer, that there should be a review process, but it should not be in such a way that it would halt development for internal contributions.  During conversations with Nirea, it was made clear that there would be no deadline, that if a PR does not receive a review, it would sit there indefinitely, never being merged in.

I'd like to see something that will encourage contribution and will also protect contributors, as a community driven project, and set a deadline, if not reviewed by X date, if the contribution is from staff, it should be allowed to be merged. The timeline i proposed was 7 days, but this could be changed to 14, for instance, but a deadline is needed or if there is ever only 1 developer, as there is now, then the project comes to an immediate halt.

Reply:
+1 to this. GitHub already does a good job of distinguishing between these two. 

Collaborator: Someone who has been invited to work directly on the project as a **member of the repository**. Has permission to push. Someone who has been actively invited to work on the core development team.

Contributor: someone who contributes something to the project, even if they are not a direct member of the repository.

These guidelines refer to Contributors (hence CONTRIBUTING.md), not Collaborators.

Reply:
There is no "approved by default if not reviewed in X days" rule.

I will not distribute unreviewed code to the inworld system.